### PR TITLE
fix(klarna): payments failing because of failed deserialization

### PR DESCRIPTION
**Pull request by Narayanbhat166**
## Type of Change

- [x] Bugfix

## Description

Klarna payments were failing because the response was not being deserialized correctly.

## Motivation and Context

During development, there was no possiblity of testing because of unavailability of authorization token for klarna.
- Remove redirection since it is not mandatory, attaching the docs for reference https://docs.klarna.com/api/payments/#operation/createOrder!c=200&path=redirect_url&t=response

## How did you test it?

Screenshot 2023-01-12 at 6 21 15 PM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code

### CODE REVIEW

**Review by SanchithHegde — DISMISSED** [review]

File: crates/router/src/core/payment_methods/transformers.rs [hunk 1]
```diff
@@ -72,10 +72,10 @@ pub fn mk_add_card_request(
     _merchant_id: &str,
 ) -> CustomResult<services::Request, errors::CardVaultError> {
     #[cfg(feature = "sandbox")]
-    let customer_id = format!("{}::{}", customer_id, _merchant_id);
+    let customer_id = &format!("{}::{}", customer_id, _merchant_id);
```
Comment (NishantJoshi00) [inline crates/router/src/core/payment_methods/transformers.rs:75]: please, rename `_merchant_id` -> `merchant_id`

**Review by NishantJoshi00 — CHANGES_REQUESTED** [review]
Other then that, LGTM

**Review by dracarys18 — APPROVED** [review]

File: crates/router/src/core/payment_methods/transformers.rs (hunk 1 above)
Comment (Narayanbhat166) [reply crates/router/src/core/payment_methods/transformers.rs]: The checks fail on CI

**Review by dracarys18 — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Review by SanchithHegde — DISMISSED** [review]

**Review by NishantJoshi00 — APPROVED** [review]
LGTM

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**